src/LaunchDarkly.ServerSdk.Redis/RedisDataStoreImpl.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using LaunchDarkly.Logging;
+using LaunchDarkly.Sdk.Server.Interfaces;
+using StackExchange.Redis;
+
+using static LaunchDarkly.Sdk.Server.Interfaces.DataStoreTypes;
+
+namespace LaunchDarkly.Sdk.Server.Integrations
+{
+    /// <summary>
+    /// Internal implementation of the Redis data store.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// Implementation notes:
+    /// </para>
+    /// <list type="bullet">
+    /// <item> Feature flags, segments, and any other kind of entity the LaunchDarkly client may wish
+    /// to store, are stored as hash values with the main key "{prefix}:features", "{prefix}:segments",
+    /// etc.</item>
+    /// <item> Redis only allows a single string value per hash key, so there is no way to store the
+    /// item metadata (version number and deletion status) separately from the value. The SDK understands
+    /// that some data store implementations don't have that capability, so it will always pass us a
+    /// serialized item string that contains the metadata in it, and we're allowed to return 0 as the
+    /// version number of a queried item to indicate "you have to deserialize the item to find out the
+    /// metadata".
+    /// </item>
+    /// <item> The special key "{prefix}:$inited" indicates that the store contains a complete data set.</item>
+    /// </list>
+    /// </remarks>
+    internal sealed class RedisDataStoreImpl : IPersistentDataStore
+    {
+        private readonly ConnectionMultiplexer _redis;
+        private readonly string _prefix;
+        private readonly Logger _log;
+        
+        // This is used for unit testing only
+        internal Action _updateHook;
+
+        internal RedisDataStoreImpl(
+            ConfigurationOptions redisConfig,
+            string prefix,
+            Logger log
+            )
+        {
+            _log = log;
+            var redisConfigCopy = redisConfig.Clone();
+            _redis = ConnectionMultiplexer.Connect(redisConfigCopy);
+            _prefix = prefix;
+            _log.Info("Using Redis data store at {0} with prefix \"{1}\"",
+                string.Join(", ", redisConfig.EndPoints.Select(DescribeEndPoint)), prefix);

```

Comment:
Here I've changed the log message slightly, with a goal of standardizing these better across SDKs.

Previously, the .NET Redis store logged this at startup:

`Creating Redis feature store using Redis server(s) at [Unspecified/localhost:6379]`

I've changed "feature store" to our new standard term "data store", got rid of the weird "Unspecified/" by not relying on the default behavior of EndPoint.ToString(), and added the prefix because it's useful and because the other SDKs log it. Also, I changed "Creating" to "Using", because the former might misleadingly imply that it's actually starting a new Redis server instance.